### PR TITLE
Must be faster

### PR DESCRIPTION
**Pull request by dustinhiatt-wf**
CODE REVIEW

Made some minor changes to reduce cache contention.  New benchmarks in documentation.

@alexandercampbell-wf @beaulyddon-wf @tannermiller-wf @ericolson-wf @rosshendrickson-wf @tylertreat @stevenosborne-wf

### CODE REVIEW

File: queue/ring.go [hunk 1]
```diff
@@ -49,8 +49,14 @@ type nodes []*node
...
-	nodes                          nodes
-	queue, dequeue, mask, disposed uint64
+	buffer0        [8]uint64
+	queue          uint64
+	buffer1        [8]uint64
+	dequeue        uint64
+	buffer2        [8]uint64
+	mask, disposed uint64
+	buffer3        [8]uint64
+	nodes          nodes
```
Comment (alexandercampbell-wf) [inline queue/ring.go]: You can use `_` in struct definitions to be explicit about padding (I've seen it used in some C bindings)

**Comment by alexandercampbell-wf** [comment]
+1

Isn't this kind of optimization highly specific to each system configuration?

**Comment by dustinhiatt-wf** [comment]
Most line sizes are 64 bytes, so it works most of the time.  We most definitely have 64-byte alignment and I think the servers we are using also have Haswells.

**Comment by tylertreat-wf** [comment]
+1

**Comment by rosshendrickson-wf** [comment]
+1

**Comment by tannermiller-wf** [comment]
+1

**Comment by beaulyddon-wf** [comment]
+1

**Pull request closed without merging**